src/GitVersionCore/GitRepoMetadataProvider.cs

```diff
@@ -105,17 +106,26 @@ public IEnumerable<Branch> GetBranchesContainingCommit([NotNull] Commit commit,
 
         /// <summary>
         /// Find the merge base of the two branches, i.e. the best common ancestor of the two branches' tips.
+        /// Note that a local branch and a remote branch tracked by it (or when untracked, has the same name) are considered the same.
         /// </summary>
         public Commit FindMergeBase(Branch branch, Branch otherBranch)
         {
-            var key = Tuple.Create(branch, otherBranch);
+            if (branch.IsSameBranch(otherBranch))
```

Comment:
@JakeGinnivan Actually here it might not be the case.

> Consider a local and its remote (tracked) branch the same, with the state of the local branch preferred.

Normally for the merge base it will make no difference if the local or remote branch is not up to date. That is only the case if the local and/or the remote branch was rebased. With this change (actually the algorithm in `IsSameBranch()`), the state (and thus merge base) of the local branch is always chosen.